src/test/groovy/com/user_service/services/UserAuthenticationContractTest.groovy

```diff
@@ -0,0 +1,148 @@
+    }
+
+    def "Should send verification code when forgot password"() {
+        given: "ForgotPasswordDto is provided"
+            ForgotPasswordDto forgotPasswordDto =
+                    new ForgotPasswordDto("jarek@gmail.com")
+
+        and: "ForgotPasswordResponseDto is provided"
+            ForgotPasswordResponseDto forgotPasswordResponseDto =
+                    new ForgotPasswordResponseDto("j***@g***.com", "email")
+
+        and: "forgotPassword method is mocked"
+            userAuthenticationService.forgotPassword(forgotPasswordDto) >> forgotPasswordResponseDto
+
+        when: "forgotPassword method is called"
+            def forgotPasswordResponse = userAuthenticationController.forgotPassword(forgotPasswordDto)
+
+        then: "expected values are returned"
+            forgotPasswordResponse.getStatusCode() == HttpStatus.OK
+            forgotPasswordResponse.getBody().email() == "j***@g***.com"
+            forgotPasswordResponse.getBody().deliveryMedium() == "email"
+    }
+
+    def "Should change password after providing code"() {
+        given: "ConfirmForgotPasswordDto is provided"
+            ConfirmForgotPasswordDto confirmForgotPasswordDto =
+                    new ConfirmForgotPasswordDto("jarek@gmail.com", "123", "dfG123!@0123456789")
+
+        and: "ConfirmForgotPasswordResponseDto is provided"
+            ConfirmForgotPasswordResponseDto confirmForgotPasswordResponseDto =
+                    new ConfirmForgotPasswordResponseDto("SUCCESS")
+
+        and: "confirmForgotPassword method is mocked"
+            userAuthenticationService.confirmForgotPassword(confirmForgotPasswordDto) >> confirmForgotPasswordResponseDto
+
+        when: "confirmForgotPassword method is called"
+            def confirmForgotPasswordResponse = userAuthenticationController.confirmForgotPassword(confirmForgotPasswordDto)
+
+        then: "expected values are returned"
+            confirmForgotPasswordResponse.getStatusCode() == HttpStatus.OK
+            confirmForgotPasswordResponse.getBody().message() == "SUCCESS"
+    }
+
+    def "Should confirm registration after providing code"() {
+        given: "ConfirmSignUpDto is provided"
+            ConfirmSignUpDto confirmSignUpDto =
+                    new ConfirmSignUpDto("123", "jarek@gmail.com")
+
+        and: "ConfirmForgotPasswordResponseDto is provided"
+            ConfirmSignUpResponseDto confirmSignUpResponseDto =
+                    new ConfirmSignUpResponseDto("CONFIRMED")
+
+        and: "confirmSignUp method is mocked"
+            userAuthenticationService.confirmSignUp(confirmSignUpDto) >> confirmSignUpResponseDto
+
+        when: "confirmSignUp method is called"
+            def confirmSignUp = userAuthenticationController.confirmSignUp(confirmSignUpDto)
+
+        then: "expected values are returned"
+            confirmSignUp.getStatusCode() == HttpStatus.OK
+            confirmSignUp.getBody().message() == "CONFIRMED"
+    }
+
```

Comment:
To increase the coverage, you should also test error scenarios. So a case when email/pass are not correct, Cognito doesn't respond, etc. We can skip those if you want for the sake of this project.